### ISSUE TITLE
Hash routing URL incorrect in build artefact

### ISSUE THREAD

**Issue by dennishendriksen** [solidjs/solid-app-router#93]
Consider the following pieces of code:
```
render(() => <Router source={hashIntegration()}><App /></Router>, document.getElementById('root'));
```

```
function App() {
  return (
    <div>
	  <Link href="/test">Test link</Link>
	  <Routes>
          <Route path="/test" element={<Test />} />
	  </Routes>
    </div>
  );
}
```
from this example [router_hash_issue.zip](https://github.com/solidjs/solid-app-router/files/8315904/router_hash_issue.zip) created with `vite build` and some custom code to inline resources in the .html.

Selecting the link results in a URL change in the browser address bar:
`file:///C:/Users/Foo/index.html#/C:/Users/Foo/index.html#/test`
Instead of the expected:
`file:///C:/Users/Foo/index.html#/test`
Hosting the .html file on a file server and accessing it through http results in the same issue.

Running the same project with `vite dev` works without any issues. Could this be an absolute/relative issue? All help would be appreciated!

**Comment by rturnq** [comment]
Thanks for the report. I am able to reproduce and it looks like the same underlying issue as #89, but this is a good alternate case. I will try to commit a fix later this evening.

**Comment by rturnq** [comment]
This is resolved in 0.3.2

**Comment by dennishendriksen** [comment]
Can confirm that 0.3.2 resolves the issue, thanks @rturnq for the quick fix!
